packages/fuels-core/src/json_abi.rs

```diff
@@ -1306,6 +1395,135 @@ mod tests {
+            }
+        ]
+        "#;
+
+        let values: Vec<String> = vec!["(42, 42)".to_string()];
+
+        let mut abi = ABIParser::new();
+
+        let function_name = "takes_tuple";
+
+        let encoded = abi
+            .encode_with_function_selector(json_abi, function_name, &values)
+            .unwrap();
+        println!("encoded: {:?}\n", encoded);
+
+        let expected_encode = "000000000e8ece68000000000000002a000000000000002a";
+        assert_eq!(encoded, expected_encode);
+    }
+
+    #[test]
+    fn nested_tuple_encode_and_decode() {
+        let json_abi = r#"
+        [
+          {
+            "type": "function",
+            "inputs": [
+              {
+                "name": "input",
+                "type": "((u64, u64), struct Person, enum State)",
+                "components": [
+                  {
+                    "name": "__tuple_element",
+                    "type": "(u64, u64)",
+                    "components": [
+                      {
+                        "name": "__tuple_element",
+                        "type": "u64",
+                        "components": null
+                      },
+                      {
+                        "name": "__tuple_element",
+                        "type": "u64",
+                        "components": null
+                      }
+                    ]
+                  },
+                  {
+                    "name": "__tuple_element",
+                    "type": "struct Person",
+                    "components": [
+                      {
+                        "name": "name",
+                        "type": "str[4]",
+                        "components": null
+                      }
+                    ]
+                  },
+                  {
+                    "name": "__tuple_element",
+                    "type": "enum State",
+                    "components": [
+                      {
+                        "name": "A",
+                        "type": "()",
+                        "components": []
+                      },
+                      {
+                        "name": "B",
+                        "type": "()",
+                        "components": []
+                      },
+                      {
+                        "name": "C",
+                        "type": "()",
+                        "components": []
+                      }
+                    ]
+                  }
+                ]
+              }
+            ],
+            "name": "takes_nested_tuple",
+            "outputs":[]
+          }
+        ]
+        "#;
+
+        let values: Vec<String> = vec!["((42, 42), (John), (1, 0))".to_string()];
+
+        let mut abi = ABIParser::new();
+
+        let function_name = "takes_nested_tuple";
+
+        let encoded = abi
+            .encode_with_function_selector(json_abi, function_name, &values)
+            .unwrap();
+        println!("encoded: {:?}\n", encoded);
```

Comment:
We probably don't want indiscriminate `println!`s anywhere

Reply:
Thanks for the input. I will delete them all.